Lib/Src/cpp_link.cpp

```diff
@@ -15,24 +17,46 @@ extern "C"
 {
 #endif
 
+#define ADC_CHANNELS 14
+
     void cpp_main()
     {
         CUartCom uart_for_errors(&huart2);
 
         CLog log_main(&uart_for_errors, "Main");
         log_main.setLogLevel(CLog::LOG_INFO);
 
+        CAdc adc_1(&hadc, ADC_CHANNELS);
+
 #ifdef DEBUG
         log_main.log(CLog::LOG_INFO, "Entered cpp_main function");
 #endif
-        log_main.log(CLog::LOG_ERROR, "Pass debugging point");
-        log_main.log(CLog::LOG_WARNING, "Hello this is a warning!");
-        // C++ Code here
 
         // Infinite Loop
         while (1)
         {
+#ifdef DEBUG
+            // Test 3 of the 14 ADC channels
+
+            // First read channels
+            uint16_t adc_ch_1 = adc_1[0];
```

Comment:
This is excellent. However... it is obscure and might be confusing to a noob. But... it is very elegant. I like it.
Not something you need to change, but think about your choices here between being elegant and writing maintainable code. Maintainable by anyone.

Reply:
To clarify, are you happy to sacrifice accessibility to the code to some people because they are very new to c++ and not as smart or experienced for sake of beautiful code. This decision you should be very wary of depending the team you are working in.

Reply:
I think I'll leave it in this case. It makes it clearer that the channels start from 0 and it is easy to use. And if someone new to C++ doesn't know about operator overloading it could be a good learning opportunity.

Reply:
I agree with your decision. The point I was trying to make is that at some stage in your career, you'll be in charge of a team. This is when you need to learn to accomplish things using someone else's skillset. If you have mixed skill team, it is useful to keep in mind that higher skill people might not be writing code with lower skill colleagues in mind. As a leader it'll be your job to make sure everyone is comfortable. You'll recognise what I am talking about when you see it in reality.